src/v2/Apps/Order/Components/PollAccountBalance.tsx

```diff
@@ -0,0 +1,139 @@
+import { FC, useState } from "react"
+import { graphql, createRefetchContainer, RelayRefetchProp } from "react-relay"
+import { Spinner } from "@artsy/palette"
+import styled from "styled-components"
+
+import { useSystemContext } from "v2/System"
+import { SystemQueryRenderer } from "v2/System/Relay/SystemQueryRenderer"
+import { renderWithLoadProgress } from "v2/System/Relay/renderWithLoadProgress" // todo
+import { usePoll } from "v2/Utils/Hooks/usePoll"
+import {
+  PollAccountBalanceQuery,
+  PollAccountBalanceQueryResponse,
+} from "v2/__generated__/PollAccountBalanceQuery.graphql"
+import { PollAccountBalance_commerceBankAccountBalance } from "v2/__generated__/PollAccountBalance_commerceBankAccountBalance.graphql"
+
+interface PollAccountBalanceProps {
+  relay: RelayRefetchProp
+  setupIntentId: string
+  commerceBankAccountBalance: PollAccountBalance_commerceBankAccountBalance | null
+  onBalanceCheckComplete: (displayInsufficientFundsError: boolean) => void
+  buyerTotalCents: number
+  orderCurrencyCode: string
+}
+
+const PollAccountBalance: FC<PollAccountBalanceProps> = ({
+  relay,
+  setupIntentId,
+  commerceBankAccountBalance,
+  onBalanceCheckComplete,
+  buyerTotalCents,
+  orderCurrencyCode,
+}) => {
+  const [shouldStopPolling, setShouldStopPolling] = useState(false)
+  const timeoutID = setTimeout(() => setShouldStopPolling(true), 5000)
+
+  usePoll({
+    callback: () => {
+      relay.refetch({ setupIntentId }, null, {}, { force: true })
+    },
+    intervalTime: 1300,
+    key: setupIntentId,
+    clearWhen: shouldStopPolling,
+  })
+
+  const userHasEnoughBalance =
+    commerceBankAccountBalance?.balanceCents &&
+    commerceBankAccountBalance?.balanceCents >= buyerTotalCents &&
+    orderCurrencyCode === commerceBankAccountBalance?.currencyCode
+
+  // we receive the balance from Stripe and it's insufficient to buy the artwork
+  const userHasInsufficientBalance =
+    commerceBankAccountBalance?.balanceCents &&
+    commerceBankAccountBalance?.balanceCents < buyerTotalCents &&
+    orderCurrencyCode === commerceBankAccountBalance?.currencyCode
```

Comment:
It's not clear to me why we are defining these two variables that seem almost identical - or how they differ from the useState `showInsufficiencyFundsError` variable above. It seems these are both either `undefined` or boolean opposites - could L51 be written as 
`const userHasInsufficientFunds = typeof userHasEnoughBalance === 'undefined' ? 'undefined' : !userHasEnoughBalance`? I'm guessing there is an even cleaner way but trying to check my understanding here.

Reply:
thanks both for the suggestions 🙏 

@erikdstock they differ from `showInsufficiencyFundsError` in that they are local variables in this component and are not accessed by any other component. they are used to determine when and with which param to fire `onBalanceCheckComplete`. I would also not rely on types here, I think it'd be even better to cast these variables as booleans specifically. 

@starsirius  the thing is that `commerceBankAccountBalance` can be null too. destructuring `balanceCents` and `currencyCode` from it would cause an error if we do it before checking `commerceBankAccountBalance`'s value (we need to check it since we'd still want to check if polling should stop already when it is null). I'll try to refactor this logic in light of that. let me know if you have any other idea!